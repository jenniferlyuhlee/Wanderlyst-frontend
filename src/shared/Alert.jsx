import React from "react";

/** Alert component to display users form submission results
 * Loops through messages and displays them
 * Props: type("danger", "success"), messages []
 */

function Alert({type, messages=[]}){
    return(
        <div>
            <ul>
                {messages.map(err => (
                    <li key={err}>{err}</li>
                ))}
            </ul>
        </div>
    )
}

export default Alert;