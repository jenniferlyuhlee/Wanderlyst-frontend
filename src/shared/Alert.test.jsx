import React from "react";
import { render } from "@testing-library/react";
import Alert from "./Alert";
import { expect, describe, test } from "vitest";

describe("Alert component", () => {
    test("renders without crashing", () => {
        render(<Alert />)
    });

    test("matches snapshot for error alerts", function() {
        let messages=["Password needs to be atleast 5 characters.", "Username taken."]
    
        const {asFragment}= render(
            <Alert type="danger" messages={messages}/>
        );
        expect(asFragment()).toMatchSnapshot();
    })

    test("matches snapshot for success alerts", function (){
        let messages=["Successful", "Saved"]
    
        const {asFragment} = render(
            <Alert type="success" messages={messages} />
        );
        expect(asFragment()).toMatchSnapshot();
    });
})