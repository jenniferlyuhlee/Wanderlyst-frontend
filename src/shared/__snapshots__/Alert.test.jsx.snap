// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert component > matches snapshot for error alerts 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger my-2"
    role="alert"
  >
    <p
      class="small m-0"
    >
      Password needs to be atleast 5 characters.
    </p>
    <p
      class="small m-0"
    >
      Username taken.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert component > matches snapshot for success alerts 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success my-2"
    role="alert"
  >
    <p
      class="small m-0"
    >
      Successful
    </p>
    <p
      class="small m-0"
    >
      Saved
    </p>
  </div>
</DocumentFragment>
`;
